### ISSUE TITLE
Unable to add custom attributes

### ISSUE THREAD

**Issue by davidojedalopez** [numical/script-ext-html-webpack-plugin#20]
Whenever I try to add a custom attribute using this plugin, the attribute just won't get added. I know the **'test'** clause is correct because I have added other attributes with that same **script matching pattern.** Also, no errors appear on the log of the build process.

I am using two **ScriptExtHtmlWebpackPlugin** declarations since I need not just the **'defer'** attribute, but also the custom one on one of the scripts.

Am I missing something? Here is my webpack.config.js:

```
const path = require("path")
const ExtractTextPlugin = require("extract-text-webpack-plugin")
const HtmlWebpackPlugin = require("html-webpack-plugin")
const CompressionPlugin = require("compression-webpack-plugin");
const ScriptExtHtmlWebpackPlugin = require("script-ext-html-webpack-plugin")
const CleanWebpackPlugin = require('clean-webpack-plugin')
const HTMLCompressionPlugin = require("html-compression-webpack-plugin");


const extractSass = new ExtractTextPlugin({
    filename: "[name].[contenthash].css",
    disable: process.env.NODE_ENV === "development"
})

let pathsToClean = [
    'dist'
]

let cleanOptions = {
    exclude: ['server.js', 'package.json', 'node_modules'],
    verbose: true,
    dry: false
}

module.exports = {
    entry: {
        index: "./src/ts/index.ts",
        intercom: "./src/js/intercom.js"
    },
    output: {
        filename: "[name].[hash].js",
        path: path.resolve(__dirname, "dist")
    },
    resolve: {
        extensions: [".webpack.js", ".web.js", ".ts", ".js"],
    },
    module: {
        rules: [{
            test: /\.(scss|css)$/,
            use: ExtractTextPlugin.extract({
                fallback: "style-loader",
                use: [{
                    loader: "css-loader",
                    options: {
                        sourceMap: false
                    },
                }, {
                    loader: "sass-loader",
                    options: {
                        sourceMap: false
                    }
                }]
            })
        }, {
            test: /\.tsx?$/,
            use: {
                loader: "ts-loader"
            }
        }, {
            test: /\.(woff2?|woff)$/,
            use: {
                loader: "url-loader"
            },
        }, {
            test: /\.(ttf|eot|svg)$/i,
            use: {
                loader: "file-loader?name=assets/[name].[ext]"
            }
        }, {
            test: /bootstrap-sass\/assets\/javascripts\//,
            use: {
                loader: 'imports-loader?jQuery=jquery'
            },
        }],
    },
    plugins: [
        new ExtractTextPlugin({
            filename: "index.[hash].css"
        }),
        new HtmlWebpackPlugin({
            template: "./src/html/index.html",
            inject: "body",
            title: 'Handy',
            filename: "index.html"
        }),
        new CompressionPlugin({
            asset: "[path].gz[query]",
            algorithm: "gzip",
            test: /\.(js|html|css)$/,
            threshold: 0,
            minRatio: 0.8
        }),
        new HTMLCompressionPlugin({
            testHTML: /\.html$/,
            asset: '[path].gz[query]',
            algorithm: 'gzip',
            threshold: 0,
            minRatio: 0.0,
            deleteOriginalAssets: false
        }),
        new ScriptExtHtmlWebpackPlugin({
            defaultAttribute: 'defer'
        }),
        new ScriptExtHtmlWebpackPlugin({
            custom: [{
                test: 'index',
                attribute: "data-turbolinks-eval",
                value: "false"
            }]
        }),
        new CleanWebpackPlugin(pathsToClean, cleanOptions)

    ],
    watch: true,
    watchOptions: {
        poll: true,
        poll: 500
    }
}
```

**Comment by numical** [comment]
Hi,
I suspect it is running two instances of ScriptExt that is causing the issue.
Try a single instance with a config on the lines of:
```
        new ScriptExtHtmlWebpackPlugin({
           sync: 'index',
           defaultAttribute: 'defer',
            custom: [{
                test: 'index',
                attribute: "data-turbolinks-eval",
                value: "false"
            }]
        }),
```

**Comment by davidojedalopez** [comment]
I just tried again with that configuration and the resulting scripts tags are as follows:

```
<script type="text/javascript" src="index.0df8a76bfb7fef3f7c82.js"></script>
<script type="text/javascript" src="intercom.0df8a76bfb7fef3f7c82.js" defer></script>
```

So, no custom attribute was added either 😕

**Comment by numical** [comment]
Oh great.  Ok I'll try to take a look over the weekend.  Sorry for your problems.

**Comment by numical** [comment]
Ok - I'm a little stumped.
I created a [test](https://github.com/numical/script-ext-html-webpack-plugin/blob/20170518-issues/spec/core-spec.js#L638) based on the bits of your config that I thought mattered and it works fine.
Which means it must be due to some part of the config that I don't think matters.

Do you have time to run webpack with the `DEBUG=ScriptExt` environmental variable set?  The output would be useful for me.  It should look something like:
```
ScriptExt html-webpack-plugin-alter-asset-tags: starting +0ms
  ScriptExt html-webpack-plugin-alter-asset-tags: replacing <head> <script> elements +1ms
  ScriptExt html-webpack-plugin-alter-asset-tags: replacing <body> <script> elements +0ms
  ScriptExt html-webpack-plugin-alter-asset-tags: processing <script> element: {"tagName":"script","closeTag":true,"attributes":{"type":"text/javascript","src":"c.3fb399d6fbb816545cae.js"}} +0ms
  ScriptExt ScriptExtHtmlWebpackPlugin: updated to: {"tagName":"script","closeTag":true,"attributes":{"type":"text/javascript","src":"c.3fb399d6fbb816545cae.js","defer":true}} +1ms
  ScriptExt html-webpack-plugin-alter-asset-tags: processing <script> element: {"tagName":"script","closeTag":true,"attributes":{"type":"text/javascript","src":"b.3fb399d6fbb816545cae.js"}} +0ms
  ScriptExt ScriptExtHtmlWebpackPlugin: updated to: {"tagName":"script","closeTag":true,"attributes":{"type":"text/javascript","src":"b.3fb399d6fbb816545cae.js","defer":true}} +1ms
  ScriptExt html-webpack-plugin-alter-asset-tags: processing <script> element: {"tagName":"script","closeTag":true,"attributes":{"type":"text/javascript","src":"a.3fb399d6fbb816545cae.js"}} +0ms
  ScriptExt ScriptExtHtmlWebpackPlugin: updated to: {"tagName":"script","closeTag":true,"attributes":{"type":"text/javascript","src":"a.3fb399d6fbb816545cae.js"}} +0ms
  ScriptExt html-webpack-plugin-alter-asset-tags: adding custom attribues to <head> <script> elements +0ms
  ScriptExt html-webpack-plugin-alter-asset-tags: adding custom attributes to <body> <script> elements +0ms
  ScriptExt ScriptExtHtmlWebpackPlugin: updated to: {"tagName":"script","closeTag":true,"attributes":{"type":"text/javascript","src":"a.3fb399d6fbb816545cae.js","data-turbolinks-eval":"false"}} +0ms
  ScriptExt html-webpack-plugin-alter-asset-tags: completed +0ms
```

**Comment by davidojedalopez** [comment]
Sorry, but I am not sure I am doing it correctly... 

I added **DEBUG=ScriptExt** to my .env file and then ran the command _webpack_. No additional logs are showing. 

Do I need to do something else?

**Comment by numical** [comment]
You simply type enter the command line:
```
DEBUG=ScripExt webpack
```
See the `package.json` file of ScriptExt for some example command lines in the `scripts` section.

Thanks for trying!

**Comment by davidojedalopez** [comment]
Awesome! I used the command:

```
DEBUG=ScriptExt VERSION=webpack2 webpack
```

And got this output:

```
ScriptExt html-webpack-plugin-alter-asset-tags: starting +0ms
  ScriptExt html-webpack-plugin-alter-asset-tags: replacing <head> <script> elements +1ms
  ScriptExt html-webpack-plugin-alter-asset-tags: replacing <body> <script> elements +0ms
  ScriptExt html-webpack-plugin-alter-asset-tags: processing <script> element: {"tagName":"script","closeTag":true,"attributes":{"type":"text/javascript","src":"index.9dd4926b1f1a4d5e779f.js"}} +1ms
  ScriptExt ScriptExtHtmlWebpackPlugin: updated to: {"tagName":"script","closeTag":true,"attributes":{"type":"text/javascript","src":"index.9dd4926b1f1a4d5e779f.js","defer":true}} +0ms
  ScriptExt html-webpack-plugin-alter-asset-tags: processing <script> element: {"tagName":"script","closeTag":true,"attributes":{"type":"text/javascript","src":"intercom.9dd4926b1f1a4d5e779f.js"}} +0ms
  ScriptExt ScriptExtHtmlWebpackPlugin: updated to: {"tagName":"script","closeTag":true,"attributes":{"type":"text/javascript","src":"intercom.9dd4926b1f1a4d5e779f.js","defer":true}} +0ms
  ScriptExt html-webpack-plugin-alter-asset-tags: completed +0ms
```

**Comment by numical** [comment]
Thanks.  Yes, your observation is puzzling.  And, in fact, the 'index' script matching pattern does not seem to be working at all as the 'index.xxx.js' file also has a `defer` attribute.
So, dependent on how much time you are willing to spend on this, here are some (fairly random) actions you could try:
1. Double check the version of ScriptExt you have - actually go into the  `node_modules/script-ext...` diectory and check the version in `package.json` - it must be 1.8.x;
2. Change your entry point names - just in case there is some weird issue with 'index';
3. Try a build with all plugins expect HtmlWebpackPlugin and ScriptEx removed - just in case one of theem is somehow interfering with the build.

**Comment by davidojedalopez** [comment]
Thanks for the recommendations! I just found that I was not using the latest plugin version 😅 

I updated to 1.8.0 version and the custom attributes now work as expected. Also, with the plugin configuration we have been using, I just get the defer attribute on the intercom script, not on the index one. Which is correct. That means that the script matching pattern is also working well. 

Thanks a lot for your help!!
